### ISSUE TITLE
'batch_size' in TestMLSTM

### ISSUE THREAD

**Issue by EnergeticChubby** [myscience/x-lstm#5]
Thanks for your fantastic work!

```python
class TestMLSTM(unittest.TestCase):
    def setUp(self):        
        self.inp_dim = 10
        self.head_dim = 8
        self.head_num = 4
        self.hid_dim = self.head_num * self.head_dim
        
        self.batch_size = 1
        
        # Create an instance of mLSTM
        self.model = mLSTM(self.inp_dim, self.head_num, self.head_dim)
        self.input = torch.randn(self.batch_size, self.inp_dim)
        
        self.hid_0 = self.model.init_hidden(self.batch_size)
```

If I set the batch_size = 1, it went wrong.

------

einops.EinopsError:  Error while processing rearrange-reduction pattern "b (h d) -> b h d".
 Input tensor shape: torch.Size([32]). Additional info: {'h': 4}.
 Wrong shape: expected 2 dims. Received 1-dim tensor.

**Comment by EnergeticChubby** [comment]
Oops, I fixed this debug. Maybe it can work!

```python
wq_xc = self.W_q(x_c)
wq_xc = wq_xc if wq_xc.dim() != 1 else wq_xc.unsqueeze(0)
wk_xc = self.W_k(x_c)
wk_xc = wk_xc if wk_xc.dim() != 1 else wk_xc.unsqueeze(0)
wv_xt = self.W_v(x_t)
wv_xt = wv_xt if wv_xt.dim() != 1 else wv_xt.unsqueeze(0)

q_t = rearrange(wq_xc, 'b (h d) -> b h d', h=self.head_num)
k_t = rearrange(wk_xc, 'b (h d) -> b h d', h=self.head_num) / sqrt(self.head_dim)
v_t = rearrange(wv_xt, 'b (h d) -> b h d', h=self.head_num)
```

```python
x_c = x_c if x_c.dim() != 1 else x_c.unsqueeze(0)
x_c = rearrange(x_c, 'b i -> b i 1')
```

Both are in the file xlstm/lstm.py .

**Comment by myscience** [comment]
Hey @EnergeticChubby! Thanks so much for spotting this out (good catch!). Indeed test code for `mLSTM` failed for `batch_size = 1`. I went through the code and the problem seemed to be that I was squeezing the tensor *assuming* batch dimension wouldn't be squeezed out as well! I replaced the `silu(x_c).squeeze()` call with an explicit reshaping that should preserve batch dimension even if trivial.  

```python
x_c = rearrange(silu(x_c), 'b ... -> b (...)') # shape: b   (i * p_factor)
```

I have pushed this fix in the repo and checked that the `mLSTM` test passes for `batch_size = 1`. Let me know if it works for you as well!
